squid/squid.conf

```diff
@@ -91,16 +91,29 @@ sslcrtd_children 8 startup=1 idle=1
 
 # --- END SSL BUMP CONFIGURATION ---
 
-# --- CACHING CONFIGURATION ---
-# Memory cache configuration for in-pod RAM caching
-cache_mem 256 MB  # Allocate 256MB of RAM for caching
+# --- STORE ID CONFIGURATION ---
+# CDN requests requiring store-id helper. Matches URLs like:
+# - https://cdn.quay.io/foo/sha256/ab/abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890
+# - https://cdn01.quay.io/foo/sha256/ab/abcdef1234567890abcdef1234567890abcdef1234567890abcdef1234567890
+acl cdns url_regex ^https://cdn([0-9]{2})?\.quay\.io/.+/sha256/.+/[a-f0-9]{64}
+# Use the store-id helper for CDN URLs
+store_id_access allow cdns
+# Don't use the store-id helper for all other URLs
+store_id_access deny all
+# The store-id helper executable
+store_id_program /usr/local/bin/squid-store-id
+# Run 1 helper process upon startup and keep at least 1 spare; scale up to 20 as needed
```

Comment:
if it can keep helper processes up for more then one request and spin up more, then perhaps if can also do out of order and async requesting so that we can use go routines and some dispatching to get more out of a single process? 

That might be premature optimization though, but I suspect we need to think about monitoring those processes to see their impact on performance since those are really deep into our caching hot path.